zilliqa/src/consensus.rs

```diff
@@ -1512,12 +1553,14 @@ impl Consensus {
                 )?;
 
                 // Return whether the head block hash changed as to whether it was new
-                Ok(self.head_block().hash() != current_head.hash())
+                let was_new = self.head_block().hash() != current_head.hash();
+
+                Ok((was_new, response))
```

Comment:
Also, is it really worth sending `Vote` messages? Any non-leader node in the committee will spam stale `Vote`s for every single block it syncs to, I think.

But really we're only ever interested in sending messages for the latest pending proposal (if any), and those messages can be either a Vote if we aren't the leader, or a Proposal iff we are the leader and buffered enough votes to accept the pending one.
So I think `Proposal`s returned will always be worth broadcasting, but `Vote`s will usually be stale.
Dropping the Vote seems like it shouldn't harm consensus too much usually (unless you're the last vote needed for a 2/3 majority and happened to be offline, which seems rare) and there's no easy way to detect whether we're at the head of the chain (and thus our vote matters) or not (and then it'd be stale). I think avoiding spamming the network with messages for every single block during on sync is probably worth dropping an occasional very rare meaningful vote.